ports/stm32/boards/Passport/modules/pages/pin_entry_page.py

```diff
@@ -258,12 +260,20 @@ def on_key(self, event):
             self.show_brick_warning()
 
     async def on_security_words(self, security_words, error):
+        from serializations import sha256
         # NOTE: Be aware that this is called from the context of another task
         if error is None:
             self.security_words = security_words
             self.show_security_words = True
             if self.show_security_words and self.user_wants_to_see_security_words:
-                self.update_message(show_security_words=True, title=self.security_words_message)
+                new_pin_sha = sha256(self.pin)
+                true_pin_sha = common.settings.get('first_four_hash')
+                if self.check_first_four and not all(x == y for x, y in zip(new_pin_sha, true_pin_sha)):
+                    self.security_words_message = ("Your first four digits are wrong.\n"
+                                                   "These are not your security words.\n"
```

Comment:
Let's not show the words at all if the hash doesn't match.  That's an attack vector for someone to just discover all the words.

Reply:
This check only happens when the user is logged in, so would this still be a security concern?

Reply:
Yes, if the device happened to be left unattended.